iceoryx_utils/test/stresstests/test_stress_sofi.cpp

```diff
@@ -0,0 +1,545 @@
+// Copyright (c) 2019 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_utils/internal/concurrent/sofi.hpp"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <mutex>
+#include <queue>
+#include <stdlib.h>
+#include <thread>
+
+using namespace testing;
+
+using SoFiData = int64_t;
+constexpr SoFiData INVALID_SOFI_DATA{-1};
+
+constexpr int64_t STRESS_TIME_HOURS{0};
+constexpr int64_t STRESS_TIME_MINUTES{0};
+constexpr int64_t STRESS_TIME_SECONDS{2};
+constexpr std::chrono::milliseconds STRESS_TIME{
+    ((STRESS_TIME_HOURS * 60 + STRESS_TIME_MINUTES) * 60 + STRESS_TIME_SECONDS) * 1000};
+
+class SoFiStress : public Test
+{
+  public:
+  protected:
+    virtual void SetUp()
+    {
+    }
+
+    virtual void TearDown()
+    {
+    }
+
+
+    /// @brief Sets the CPU affinity for a thread
+    ///
+    /// @param cpu is the CPU the thread shall use
+    /// @param nativeHandle is the native handle of the c++11 std::thread
+    /// @return bool: if true, setting the affinity was successfull
+    bool setCpuAffinity(unsigned int cpu, std::thread::native_handle_type nativeHandle)
+    {
+#ifndef __QNX__
```

Comment:
I think we have to be more specific since CPU_SETs are a linux only feature which will not work on any other platform